File: C++/Bit Manipulation/1318. Minimum Flips to Make a OR b Equal to c.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int minFlips(int a, int b, int c) {
        int result = 0;
        while (a || b || c) {
            int bitA = a & 1, bitB = b & 1, bitC = c & 1;
            if (bitC) result += (1 - (bitA | bitB));
            else result += 2 - !bitA - !bitB;
            a >>= 1, b >>= 1, c >>= 1;
        }
        return result;
    }
};

File: Python/bit manipulation/1318. Minimum Flips to Make a OR b Equal to c.py
class Solution:
    def minFlips(self, a: int, b: int, c: int) -> int:
        res = 0
        while a or b or c:
            bit_a, bit_b = a & 1, b & 1,
            if c & 1:
                res += 0 if bit_a or bit_b else 1
            else:
                res += (bit_a + bit_b)
            a, b, c = a >> 1, b >> 1, c >> 1
        return res
